include/boost/di.hpp

```diff
@@ -1738,11 +1738,7 @@ class dependency
 #if defined(__cpp_variable_templates)
   dependency& operator()() noexcept { return *this; }
 #endif
-#if defined(__MSVC__)
  public:
```

Comment:
Hmm, I remember issues with it in MSV-2015?

Reply:
The problem here is that while this part is protected there is no way to extend dependency. That why I opened it so that anyone can call it. Actually it was open for VS I just expanded it for other compilers